File: app/models/Topic.py
from .ModelData import ModelData
from .FirebaseModel import check_document

class Topic(ModelData):
    def __init__(self, **kwargs):
        self.dir = ""
        self.token = ""
        super().__init__(**kwargs)
    
    def show(self):
        return f"{self.title} : {self.info}"
    
    def check_token(self):
        return check_document(self.token)

File: src/models/Topic.ts
export interface ITopicProperties {
  title: string;
  description: string;
  version: string;
  token: string;
  code?: string;
  image?: string;
}

export default class Topic implements ITopicProperties {
  title: string;
  description: string;
  version: string;
  token: string;
  code?: string;
  image?: string;

  constructor(data: ITopicProperties) {
    this.title = data.title;
    this.description = data.description;
    this.version = data.version;
    this.token = data.token;
    this.code = data.code;
    this.image = data.image;
  }
}
